sources/packages/api/src/services/student/student.service.ts

```diff
@@ -143,27 +149,28 @@ export class StudentService extends RecordDataModelService<Student> {
         "Not able to find a student using the username (bcsc name)",
       );
     }
-
     let mustSave = false;
-
+    if (userToken.givenNames === undefined) {
+      userToken.givenNames = null;
+    }
     if (
-      userToken.email !== studentToSync.user.email ||
+      !dayjs(userToken.birthdate).isSame(studentToSync.birthDate) ||
       userToken.lastName !== studentToSync.user.lastName ||
       userToken.givenNames !== studentToSync.user.firstName
     ) {
-      studentToSync.user.email = userToken.email;
+      const sinValidation = new SINValidation();
+      studentToSync.birthDate = getDateOnly(userToken.birthdate);
       studentToSync.user.lastName = userToken.lastName;
       studentToSync.user.firstName = userToken.givenNames;
+      sinValidation.user = studentToSync.user;
+      studentToSync.sinValidation = sinValidation;
       mustSave = true;
-      //TODO update the SIN validation here
     }
-
-    const userTokenBirthdate = getDateOnly(userToken.birthdate);
     if (
-      userTokenBirthdate !== studentToSync.birthDate ||
+      userToken.email !== studentToSync.user.email ||
```

Comment:
These conditions can be added to the previous if itself. why has it been added here separately?

Reply:
Is it because, change in gender and email will not trigger SIN validation again right?